### ISSUE TITLE
Question/Support needed: How to use a SSL-secured database connection

### ISSUE THREAD

**Issue by FreshMosh** [photoprism/photoprism#1452]
Hey folks, 
I'm not able to establish a SSL-enabled database connection. 

Setup is: 
MySQL 8.x Server provided by hosting/cloud provider. 
Docker Swarm Cluster on VM's at the same hosting/cloud provider

When I set the MySQL-server on "enforce SSL" Photoprism cant connect anymore. 

Adding this to the DSN: 
`sslmode=required`
did not do anything. 

The SSL-certificate used by the MySQL-server is issued by a trusted global CA and thus trusted by default as soon as ca-certificates (ubuntu) is installed. Which is the case for the photoprism docker image. 

I also didn't find any documentation on this topic from Photoprism. 

Regards

**Comment by lastzero** [comment]
Since local / Docker internal database connections typically don't require encryption, we honestly have no experience with that. Also note encryption has a performance impact, not sure to what extent. In more than two decades for projects large and small, I've never used SSL with MySQL.

**Comment by lastzero** [comment]
https://stackoverflow.com/questions/42163732/how-to-use-the-go-mysql-driver-with-ssl-on-aws-with-a-mysql-rds-instance

**Comment by FreshMosh** [comment]
Neither did I. Yet I'm currently in this exact situation. 

Thanks for the link. I was able to find the missing link through this. 
Adding `tls=true` to the DSN leads to the desired behaviour.
https://pkg.go.dev/github.com/go-sql-driver/mysql?utm_source=godoc#readme-tls

More or less. 

After a bit of more digging it seems like this could ultimately be an issue with the providers certificate. 
While debugging this issue I got this error message: 
`level=fatal msg="x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0"`
which led me to this:
https://github.com/rancher/rke2/issues/775#issuecomment-797285601

> This is not a defect in go, it is a feature. The check was added in 1.11, enabled by default in 1.15, and the opt-out will be removed in a future release. You need to update the certificate on the registry to correct the attributes.


It is working for now using `tls=skip-verify` and the provider has an issue. 
Thanks for the hint. 

**Turns out this has nothing to do with Photoprism at all.**
